File: config/config-sbc2ha/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/action/CoverActionConfig.java
package com.dfi.sbc2ha.config.sbc2ha.definition.action;

import com.dfi.sbc2ha.config.sbc2ha.definition.enums.ActionCoverType;
import com.dfi.sbc2ha.config.sbc2ha.definition.enums.ActionType;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
public class CoverActionConfig extends ActionConfig {

    String pin;
    @JsonProperty("action_cover")
    ActionCoverType actionCover;

    public CoverActionConfig() {
        super();
        action = ActionType.COVER;
    }
}

File: config/editor/src/app/definition/action/CoverActionConfig.ts
import {ActionConfig} from "./ActionConfig";
import {ActionCoverType} from "../enums/ActionCoverType";
import {ActionType} from "../enums/ActionType";
import {Expose} from "class-transformer";

export class CoverActionConfig extends ActionConfig {
    output: number;
    @Expose({name: "action_cover"})
    actionCover: ActionCoverType;


    constructor(output: number, actionCover: ActionCoverType) {
        super(ActionType.COVER);
        this.output = output;
        this.actionCover = actionCover;
    }
}
